wordpress-mobile/WordPress-Android
# Make number suffixes translatable through an injected class

Fixes #10753

This PR removes the `toFormattedString` extension function and replaces it with a call on an injected `StatsUtils` object. The extension function doesn't make sense since we have to use other injected fields to build the formatted string (resource provider and locale manager). 

To test:
* Check that the Stats still work
* Check that big numbers are still being shortened (15,000 -> 15k) 

PR submission checklist:
